Last-Update: 2012-11-01
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Forwarded: not-needed
Bug-Zabbix: https://support.zabbix.com/browse/ZBX-4986
Description: fix columns sorting and pagination in frontend-php
 Problem is manifesting with nginx/php-fpm
 or with Apache/mod_proxy.

Index: zabbix-2.0.4/frontends/php/include/func.inc.php
===================================================================
--- zabbix-2.0.4.orig/frontends/php/include/func.inc.php
+++ zabbix-2.0.4/frontends/php/include/func.inc.php
@@ -1538,7 +1538,7 @@ function make_sorting_header($obj, $tabf
 		$script = "javascript: return updater.onetime_update('".ZBX_PAGE_MAIN_HAT."', '".$url."');";
 	}
 	else {
-		$script = 'javascript: redirect("'.$url.'");';
+		$script = 'javascript: redirect("'.basename($url).'");';
 	}
 
 	zbx_value2array($obj);
@@ -1638,13 +1638,13 @@ function getPagingLine(&$items) {
 		$url = new Curl();
 		if ($startPage > 1) {
 			$url->setArgument('page', 1);
-			$pageline[] = new CLink('<< '._x('First', 'page navigation'), $url->getUrl(), null, null, true);
+			$pageline[] = new CLink('<< '._x('First', 'page navigation'), basename($url->getUrl()), null, null, true);
 			$pageline[] = '&nbsp;&nbsp;';
 		}
 
 		if ($currentPage > 1) {
 			$url->setArgument('page', $currentPage - 1);
-			$pageline[] = new CLink('< '._x('Previous', 'page navigation'), $url->getUrl(), null, null, true);
+			$pageline[] = new CLink('< '._x('Previous', 'page navigation'), basename($url->getUrl()), null, null, true);
 			$pageline[] = ' | ';
 		}
 
@@ -1658,7 +1658,7 @@ function getPagingLine(&$items) {
 			}
 			else {
 				$url->setArgument('page', $p);
-				$pagespan = new CLink($p, $url->getUrl(), null, null, true);
+				$pagespan = new CLink($p, basename($url->getUrl()), null, null, true);
 			}
 
 			$pageline[] = $pagespan;
@@ -1671,14 +1671,14 @@ function getPagingLine(&$items) {
 			$pageline[] = ' | ';
 
 			$url->setArgument('page', $currentPage + 1);
-			$pageline[] = new CLink(_x('Next', 'page navigation').' >', $url->getUrl(), null, null, true);
+			$pageline[] = new CLink(_x('Next', 'page navigation').' >', basename($url->getUrl()), null, null, true);
 		}
 
 		if ($p < $pagesCount) {
 			$pageline[] = '&nbsp;&nbsp;';
 
 			$url->setArgument('page', $pagesCount);
-			$pageline[] = new CLink(_x('Last', 'page navigation').' >>', $url->getUrl(), null, null, true);
+			$pageline[] = new CLink(_x('Last', 'page navigation').' >>', basename($url->getUrl()), null, null, true);
 		}
 
 		$table = new CTable(null, 'paging');
